### PR TITLE
Change design of confirmation modals in web UI

### PR DESCRIPTION
**Pull request by Gargron**
image

___

Fixes MAS-257

### CODE REVIEW

**Comment by ClearlyClaire** [comment]
From a design perspective, this looks good to me. This appear to not touch the reblog confirmation modal, is that intended?

**Comment by Gargron** [comment]
Yes, that is a different modal, it needs updating too (see Linear ticket) but it’s also got a unique design.

File: app/javascript/mastodon/features/status/containers/detailed_status_container.js [hunk 1]
```diff
@@ -50,20 +41,13 @@ const makeMapStateToProps = () => {
...
       if (state.getIn(['compose', 'text']).trim().length !== 0) {
-        dispatch(openModal({
-          modalType: 'CONFIRM',
-          modalProps: {
-            message: intl.formatMessage(messages.replyMessage),
-            confirm: intl.formatMessage(messages.replyConfirm),
-            onConfirm: () => dispatch(replyCompose(status)),
-          },
-        }));
+        dispatch(openModal({ type: 'CONFIRM_REPLY', status }));
```
Comment (ClearlyClaire) [inline app/javascript/mastodon/features/status/containers/detailed_status_container.js:50]: Shouldn't this be?
```suggestion
        dispatch(openModal({ type: 'CONFIRM_REPLY', modalProps: { status } }));
```

But looking more at the file I think the whole file may be dead code?

File: app/javascript/mastodon/features/status/containers/detailed_status_container.js [hunk 2]
```diff
@@ -100,14 +84,7 @@ const mapDispatchToProps = (dispatch, { intl }) => ({
...
     } else {
-      dispatch(openModal({
-        modalType: 'CONFIRM',
-        modalProps: {
-          message: intl.formatMessage(withRedraft ? messages.redraftMessage : messages.deleteMessage),
-          confirm: intl.formatMessage(withRedraft ? messages.redraftConfirm : messages.deleteConfirm),
-          onConfirm: () => dispatch(deleteStatus(status.get('id'), withRedraft)),
-        },
-      }));
+      dispatch(openModal({ type: 'CONFIRM_DELETE_STATUS', statusId: status.get('id'), withRedraft }));
```
Comment (ClearlyClaire) [inline app/javascript/mastodon/features/status/containers/detailed_status_container.js:87]: Same remark as above.

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]
Looks fine to me except for two errors that look like they may actually be in dead code.

File: app/javascript/mastodon/features/status/containers/detailed_status_container.js (hunk 1 above)
Comment (renchap) [reply app/javascript/mastodon/features/status/containers/detailed_status_container.js]: Pushed a fix, but yes this seems to be dead code.

I can make another PR once this one is merged to remove it.

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**